### ISSUE TITLE
Question: changing task priority rules and rebuilding tasks

### ISSUE THREAD

**Issue by govvin** [osmlab/maproulette3#999]
When you change the rules for task prioritization, are you also supposed to rebuild the task?

I tried doing that for this [challenge](https://maproulette.org/challenge/9413), to prioritize areas hit by a recent typhoon but it didn't seem like it changed the task prioritization. 

If that isn't the case, how should I update a challenge to apply changes in prioritization rules?

**Comment by nrotstan** [comment]
The priority rules are applied when you save your edits to the challenge. I suppose it's possible that for large challenges it could take a minute, so perhaps the updated rules aren't instantly reflected, but refreshing the page after a short wait should show any changes.

Can you provide a couple examples of tasks that aren't being prioritized as you expect?

**Comment by govvin** [comment]
Hi @nrotstan . In this [challenge](https://maproulette.org/challenge/9413) the highest priority for all tasks whose property is "addr:property" = Catandanues" but I don't get the expected results

![image](https://user-images.githubusercontent.com/5821688/70158293-f74a6a00-16f1-11ea-9e0c-3efd8c507dc4.png)

I know that I still have available tasks that meets the priority rules, but they are not being served first, as expected.

**Comment by nrotstan** [comment]
I think it may be because the tasks are also matching your medium priority rules (`hrslph100m` equals `true`) and are ending up with medium priority as a result.

**Comment by govvin** [comment]
I see. So the prioritization processing is done from high to low?

My initial understanding is that rules to set high priority tasks will override lower rules.

Wouldn't that be a better behavior to expect?

**Comment by nrotstan** [comment]
After looking at the code, you are right, the higher priority should be chosen over the medium priority. Can you attach the GeoJSON you're using so I can dig into it further?

**Comment by govvin** [comment]
Thank you @nrotstan . Here's an older [geojson](https://github.com/osmlab/maproulette3/files/3722622/geojson-taskings.zip), but should be similar.

**Comment by nrotstan** [comment]
I've confirmed this is due to a bug. Fixed by PR maproulette/maproulette2#652

**Comment by nrotstan** [comment]
This should now be fixed on maproulette.org

**Comment by govvin** [comment]
Hi again @nrotstan .  Thank you for taking a stab at this.

I tested this yesterday and today, and it's still not working as expected for the same challenge. 

I set a high priority for tasks that meet `addr:province`=`Batanes` but I don't see any high priority tasks being served although the dashboard says there are at least 17 tasks that meet the criteria.

Following your earlier instructions, I am only saving the priority settings changes but not rebuilding the task.

**Comment by nrotstan** [comment]
When I go to your challenge, @govvin , I get served a task in Batanes province, so it seems to be working correctly. On a challenge of that size it can take 5-10 mins for the tasks to get reprioritized.

Also, one thing I noticed in the test data you offered is that you have your own `priority` property on the tasks and it was set to `medium` on the (high priority) task I happened to receive while testing, which was a bit confusing at first. But those properties come from your GeoJSON data and are unrelated to the priorities assigned by MapRoulette. The task I just got from maproulette.org had a blank `priority`, so I'm not sure if you're using those in the real challenge or not.

**Comment by govvin** [comment]
Hi Neil. Thank you for the quick response.

I confirm that it now works as expected, and I suppose I should've waited longer before I posted the feedback. 

re priority
Yes, the geojson came with a default priority based on a pre-processing criteria (we used  [HRSL](https://ciesin.columbia.edu/data/hrsl/) data to test accuracy of the location data).  My apologies for not elaborating there. The priority key in the geojson  is not being used by MR, no?

We like the idea of having the flexibility of changing priorities on-the-fly (almost, but now I'll keep in mind to give it time) so that we can use it in activities where a geographic region may have a temporary higher priority (e.g. we are doing a mapathon there, or we are complementing another mapping activity for that region, etc.). At the end of the activity, or if we complete the temporary priorities, we switch back to our "normal" tasking criteria. 

Hence, the highest priority in the geojson is only set to medium, so we can set a higher priority for temporal events.
